rclcpp/src/rclcpp/executor.cpp

```diff
@@ -331,6 +331,32 @@ Executor::execute_subscription(
       rcl_reset_error();
     }
     subscription->return_serialized_message(serialized_msg);
+  } else if (subscription->can_loan_messages()) {
+    void * loaned_msg = nullptr;
+    auto ret = rcl_take_loaned_message(
+      subscription->get_subscription_handle().get(),
+      &loaned_msg,
+      &message_info,
+      nullptr);
+    if (RCL_RET_OK == ret) {
+      subscription->handle_loaned_message(loaned_msg, message_info);
+    } else if (RCL_RET_SUBSCRIPTION_TAKE_FAILED != ret) {
+      RCUTILS_LOG_ERROR_NAMED(
+        "rclcpp",
+        "take_loaned failed for subscription on topic '%s': %s",
+        subscription->get_topic_name(), rcl_get_error_string().str);
+      rcl_reset_error();
+    }
+    ret = rcl_release_loaned_message(
```

Comment:
`release`? Why not `return`?

Reply:
I thought we agreed on having `borrow/publish/return` on the publishing side and `take/release` on the subscription side.
At least on rmw level we need a separation as we can't overload functions and have `rmw_return_loaned_message` take either a `rmw_subscription_t` or `rmw_publisher_t`.

Reply:
Oh I remember that now, but I thought we went with `rmw_return_loaned_publisher_message()` and `rmw_return_loaned_subscription_message()` (or something similar)?

Reply:
I think we were also mentioning once having `discard_loaned_message` on the publisher side and then have `return_loaned_message` for subscriptions. 
I guess we should settle on the names soonish, as this will bring just another set of PRs with it ;-)

Reply:
I propose we stick with the suggestion of 
`rmw_return_loaned_message_from_publisher` and `rmw_return_loaned_message_from_subscription`

Reply:
Sounds good thanks.